### PR TITLE
tests: add system tests for `to_arrow` with extreme values

### PR DESCRIPTION
**Pull request by tswast**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [x] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/python-bigquery/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [x] Ensure the tests and linter pass
- [x] Code coverage does not decrease (if any source code was changed)
- [x] Appropriate docs were updated (if necessary)

Towards #812  🦕

### CODE REVIEW

File: google/cloud/bigquery/_pandas_helpers.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ def pyarrow_numeric():
 
 
 def pyarrow_bignumeric():
+    # 77th digit is partial.
+    # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_type
```
Comment (plamut) [inline google/cloud/bigquery/_pandas_helpers.py:97]:
```suggestion
    # https://cloud.google.com/bigquery/docs/reference/standard-sql/data-types#decimal_types
```

The anchor in the URL seems incorrect, it just positions you at the top of the page.

File: tests/data/scalars_extreme.jsonl [hunk 2]
```diff
@@ -0,0 +1,4 @@
+{"bool_col": true, "bytes_col": "abcd", "date_col": "9999-12-31", "datetime_col": "9999-12-31 23:59:59.999999", "geography_col": "POINT(-122.0838511 37.3860517)", "int64_col": "9223372036854775807", "numeric_col": "9.9999999999999999999999999999999999999E+28", "bignumeric_col": "9.999999999999999999999999999999999999999999999999999999999999999999999999999E+37", "float64_col": "+inf", "string_col": "Hello, World", "time_col": "23:59:59.99999", "timestamp_col": "9999-12-31T23:59:59.999999Z"}
+{"bool_col": false, "bytes_col": "abcd", "date_col": "0001-01-01", "datetime_col": "0001-01-01 00:00:00", "geography_col": "POINT(-122.0838511 37.3860517)", "int64_col": "-9223372036854775808", "numeric_col": "-9.9999999999999999999999999999999999999E+28", "bignumeric_col": "-9.999999999999999999999999999999999999999999999999999999999999999999999999999E+37", "float64_col": "-inf", "string_col": "Hello, World", "time_col": "00:00:00", "timestamp_col": "0001-01-01T00:00:00.000000Z"}
```
Comment (plamut) [inline tests/data/scalars_extreme.jsonl:2]: Is this POINT really extreme? I would expect to test at least one of the poles and some point just east of Fiji - if it matters?

**Review by plamut — COMMENTED** [review]
Looks good, and data files are neat - cleaner (IMO at least) than a large wall of parametrized test inputs.

Made two remarks, but that's it.

File: tests/data/scalars_extreme.jsonl (hunk 2 above)
Comment (tswast) [reply tests/data/scalars_extreme.jsonl]: Good idea! I've updated it to use the north and south pole. Something along the international date line would be good too, but I'd also like to have a null value. Maybe I'll add another row.

Comment (plamut) [reply tests/data/scalars_extreme.jsonl]: > Something along the international date line would be good too

Yeah, that's the "near Fiji" point. :)

**Review by plamut — APPROVED** [review]
This now looks _extremely_ good. :upside_down_face:

**Pull request merged**